meta/Filter.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace dokuwiki\plugin\struct\meta;
+
+use dokuwiki\Form\Form;
+
+/**
+ * Struct filter class
+ */
+class Filter
+{
+    protected $renderer;
+
+    /** @var \dokuwiki\plugin\struct\meta\Search */
+    protected $search;
+
+    /** @var Value[][] */
+    protected $result;
+
+    /**
+     * @param \Doku_Renderer $renderer
+     * @param \dokuwiki\plugin\struct\meta\Search $search
+     */
+    public function __construct($renderer, $search)
+    {
+        $this->renderer = $renderer;
+        $this->search = $search;
+        $this->result = $search->execute();
+    }
+
+    /**
+     * Render the filter form.
+     * Reuses the structure of advanced search tools to leverage
+     * the core grouping styles and scripts.
+     *
+     * @param array $lang Language strings "title", "intro" and "button"
+     * @return void
+     */
+    public function render($lang)
+    {
+        $schemas = $this->search->getSchemas();
+        $schema = $schemas[0]->getTable();
+
+        $colValues = $this->getAllColumnValues();
+
+        $form = new Form(['method' => 'get'], true);
+        $form->addClass('struct-filter-form search-results-form');
+        $form->setHiddenField('id', getID());
+
+        $form->addFieldsetOpen()->addClass('struct-filter-form search-form');
+        $form->addHTML('<legend>' . $lang['title'] . '</legend>');
+        $form->addHTML('<p>' . $lang['intro'] . '</p>');
+
+        $form->addTagOpen('div')
+            ->addClass('advancedOptions');
+
+        // column dropdowns
+        foreach ($colValues as $colName => $colData) {
+            $qualifiedColName = $colName[0] !== '%' ? "$schema.$colName" : $colName;
+            $values = array_unique($colData['values']);
```

Comment:
I think getAllColumnValues() should return unique values already.

Reply:
Done.